### PR TITLE
docs(auth-kit): Update docs for the new `SafeAuthPack`

### PR DESCRIPTION
**Pull request by yagopv**
This PR updates the docs for the `auth-kit`:

- https://github.com/safe-global/safe-docs/issues/262
- https://github.com/safe-global/safe-docs/issues/263

### CODE REVIEW

File: safe-core-sdk/auth-kit/safe-auth.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+### Install dependencies
+
+```bash
+npm i @safe-global/auth-kit @web3auth/safeauth-embed
+```
+
+## Create a SafeAuthPack instance
+
+We are going to use the provided `SafeAuthPack` exported in the `@safe-global/auth-kit` package.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/safe-auth.md:19]:
```suggestion
We will use the provided `SafeAuthPack` exported in the `@safe-global/auth-kit` package.
```

File: safe-core-sdk/auth-kit/safe-auth.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+```typescript
+const safeAuthPack = new SafeAuthPack({
+  txServiceUrl: 'https://safe-transaction-mainnet.safe.global',
+});
+```
+
+## Signing and executing transactions using the SafeAuthPack and Protocol Kit
+
+The `SafeAuthPack` can be used in conjunction with the [Protocol Kit](../protocol-kit/README.md) to establish a connection to a Safe. This connection is made using the `provider` and `signer` associated with the authenticated account.
```
Comment (tanay1337) [inline safe-core-sdk/auth-kit/safe-auth.md:110]:
```suggestion
The `SafeAuthPack` can be used with the [Protocol Kit](../protocol-kit/README.md) to establish a connection to a Safe. This connection is made using the `provider` and `signer` associated with the authenticated account.
```

**Review by tanay1337 — CHANGES_REQUESTED** [review]
The docs structure has changed, so the location of the files and the placement in the sidebar will also need to be updated.

File: reference/safe-core-sdk/auth-kit/SafeAuthPack.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+The `SafeAuthPack` class is what makes SafeAuth modal and Safe accounts work together. Create an instance of the pack and initialize it to start the interaction.
+
+```typescript
+const safeAuthPack = new SafeAuthPack({
+  txServiceUrl: 'https://safe-transaction-mainnet.safe.global',
+});
+await safeAuthPack.init(safeAuthInitOptions);
+```
+
+### new SafeAuthPack(safeAuthConfig)
```
[the remark this answers is not in the document]
Comment (dasanra) [reply reference/safe-core-sdk/auth-kit/SafeAuthPack.md:28]: Here we should use backticks, as we do at MoneriumPack reference. Will take care of that 👍

**Comment by dasanra** [comment]
@tanay1337 thank you for your review! I took all the suggestions and adapted the PR to match the current Safe docs structure. We can continue here:

https://github.com/safe-global/safe-docs/pull/289

**Pull request closed without merging**